### ISSUE TITLE
The header of the site changes the color when scrolling up - issue reproducible on FF Nightly 

### ISSUE THREAD

**Issue by ioanarusiczki** [mozilla/addons-frontend#11576]
STR:
On FF 103.0a1 -> open addons.mozilla.org , scroll down and then up -> Observe the header

![Nightly](https://user-images.githubusercontent.com/33448286/172839258-b6596c3d-a8d6-4526-b296-ce1a8bb3a409.png)

Or go to a detail page of an add-on and click see all versions 
Then click back and scroll up towards the header

https://user-images.githubusercontent.com/33448286/172839349-38c62df4-c162-4ee1-8d44-d2d44acf5596.mp4

Notes:
Also reproducible on Android devices with FF Nightly 
Not reproducible on release versions , currently FF101 for desktop -> we'll try to find the regression window and get back with an update.
I could not reproduce this on the blog with FF Nightly, only on AMO.
Not reproducible on Chrome.

**Comment by acornestean** [comment]
I performed a bisection to find the potential regressor and came up with: https://bugzilla.mozilla.org/show_bug.cgi?id=1731136, ,more specifically, the changes implemented with Part 5 of the bug.

The full mozregression log, in case it's relevant:

2022-06-09T14:46:12.577000: DEBUG : Found commit message:
Bug 1731136 Part 5: Make macOS native compositor and NativeLayerCA handle backdrop r=mstange
layers.

This makes the macOS native compositor claim the ability to handle color
layers and makes NativeLayerCA actually do it. Color layers have a different
structure than other layers. Color layers apply the color to the
wrappingCALayer, and have no contentCALayer at all. That means that the
color layers are always sized to the layer's clip rect.

This also contains a drive-by fix to handle the case where an mOpaquenessTintLayer
exists when mMutatedSpecializeVideo is set to true. Before this change, in such a
case, the opaqueness layer will not be associated with the recreated wrapping layer.

Depends on D128131

Differential Revision: https://phabricator.services.mozilla.com/D128545

2022-06-09T14:46:12.577000: DEBUG : Did not find a branch, checking all integration branches
2022-06-09T14:46:12.581000: INFO : The bisection is done.
2022-06-09T14:46:12.584000: INFO : Stopped

The corresponding pushlog: https://hg.mozilla.org/integration/autoland/pushloghtml?fromchange=d230cd9994f91937c70b029a57f9aefd6120b7e8&tochange=ebe5a430029fb71991ea327ded3d96de0c5b98e6
